### DISCUSSION TITLE
Lazy Load

### DISCUSSION THREAD

**Discussion by hawklorry** [DHI/mikeio1d#30] [Q&A] [answered]
Could you please provide some details on LazyLoading? 

![image](https://user-images.githubusercontent.com/3594975/173416190-0feaaefd-cb75-4e10-9ffb-7358a6df213d.png)

**Comment by JesperGr** [comment, marked as answer]
Lazy loading can be used if you only need to retrieve data for a few time steps, but for all the network. 

In the cases where time series for the entire simulation is requested at only a few locations, lazy loading is a bad idea. Use filtering instead.

This lazy loading is mostly used (and designed for use) by the engine when reading the state from a result file (hot-starting on a file)

**Reply by hawklorry** [reply]
Thanks @JesperGr for the explanation. Could you pleas also suggest the best way to get statistics of an item at all locations, for example, max water level at nodes?

**Reply by Prebbish** [reply]
Would love that as well!
